cpu/atmega_common/periph/pwm.c

```diff
@@ -58,6 +60,38 @@ static inline unsigned get_prescaler(pwm_t dev, uint32_t *scale)
     return pre;
 }
 
+static inline void compute_cra_and_crb(pwm_t dev, uint8_t pre)
+{
+    uint8_t cra = _BV(WGM1) | _BV(WGM0);
+    uint8_t crb = pre;
+
+    if (pwm_conf[dev].pin_ch[0] != GPIO_UNDEF) {
+        cra |= _BV(COMA1);
+    }
+    else {
+        crb |= _BV(WGM2);
+    }
+
+    if (pwm_conf[dev].pin_ch[0] != GPIO_UNDEF) {
+        cra |= _BV(COMB1);
+    }
+
+    state[dev].CRA = cra;
+    state[dev].CRB = crb;
+}
+
+static inline void apply_config(pwm_t dev)
+{
+    pwm_conf[dev].dev->CRA = state[dev].CRA;
+    pwm_conf[dev].dev->CRB = state[dev].CRB;
+
+    if (pwm_conf[dev].pin_ch[0] == GPIO_UNDEF) {
+        /* If channel 0 is not used, variable resolutions can be used for
+         * channel 1 */
+        pwm_conf[dev].dev->OCR[0] = state[dev].res;
+    }
+}
+
 uint32_t pwm_init(pwm_t dev, pwm_mode_t mode, uint32_t freq, uint16_t res)
```

Comment:
`mode` is unused if `DEVELHELP = 0` is set.

Reply:
Fixed